imports/api/questions.js

```diff
@@ -195,84 +197,56 @@ if (Meteor.isServer) {
     } else this.ready()
   })
 
-  // question library for a user (student can see those they own or created)
-  Meteor.publish('questions.library', function () {
+  Meteor.publish('questions.library', function (courseId) {
     if (this.userId) {
       const user = Meteor.users.findOne({_id: this.userId})
-      if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else if (user.isInstructorAnyCourse()) {
-        const courses = _.pluck(Courses.find({instructors: this.userId}).fetch(), '_id')
-        return Questions.find({
-          '$or': [{owner: this.userId}, {courseId: { '$in': courses }, approved: true}],
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else {
-        // students. By checking for creator, they can see the questions they submitted
-        // that have been moved to a course library (which changes the owner w/o copying)
-        return Questions.find({
-          '$or': [{creator: this.userId}, {owner: this.userId}],
-          sessionId: {$exists: false} })
+      const course = Courses.findOne({ _id: courseId})
```

Comment:
I think we also need to make it that anyone cannot see a private question if they are not the owner.